internal/pkg/policy/policy_output.go

```diff
@@ -246,6 +272,80 @@ func (p *Output) prepareElasticsearch(
 	return nil
 }
 
+func (p *Output) createAndGetBulker(mainBulker bulk.Bulk, outputMap map[string]map[string]interface{}) (bulk.Bulk, error) {
+	mainBulker.CheckRemoteOutputChanged(p.Name, outputMap[p.Name])
+	bulker := bulkerMap[p.Name]
+	if bulker != nil {
+		return bulker, nil
+	}
+	bulkCtx, bulkCancel := context.WithCancel(context.Background())
+	defer bulkCancel()
+	es, err := p.createRemoteEsClient(bulkCtx, outputMap)
+	if err != nil {
+		return nil, err
+	}
+	// starting a new bulker to create/update API keys for remote ES output
+	newBulker := bulk.NewBulker(es, mainBulker.Tracer())
+	bulkerMap[p.Name] = newBulker
+
+	errCh := make(chan error)
+	go func() {
+		runFunc := func() (err error) {
+			return newBulker.Run(bulkCtx)
+		}
+
+		errCh <- runFunc()
+	}()
+	go func() {
+		select {
+		case err = <-errCh:
+		case <-bulkCtx.Done():
+			err = bulkCtx.Err()
+		}
+	}()
+
+	return newBulker, nil
+}
+
+func (p *Output) createRemoteEsClient(ctx context.Context, outputMap map[string]map[string]interface{}) (*elasticsearch.Client, error) {
+	hostsObj := outputMap[p.Name]["hosts"]
+	hosts, ok := hostsObj.([]interface{})
+	if !ok {
+		return nil, fmt.Errorf("failed to get hosts from output: %v", hostsObj)
+	}
+	hostsStrings := make([]string, len(hosts))
+	for i, host := range hosts {
+		hostsStrings[i], ok = host.(string)
+		if !ok {
+			return nil, fmt.Errorf("failed to get hosts from output: %v", host)
+		}
+	}
+
+	cfg := config.Config{
+		Output: config.Output{
+			Elasticsearch: config.Elasticsearch{
+				Hosts:        hostsStrings,
+				ServiceToken: p.ServiceToken,
+			},
+		},
+	}
+	es, err := es.NewClient(ctx, &cfg, false, elasticsearchOptions(
```

Comment:
If this fails does Fleet server fail to start? If so this is another point of system failure.

Most users will not want their remote monitoring ES output being unavailable to cause all of their Fleet Servers and Agents to go offline. The system should degrade gracefully in this case and treat the remote ES output as a nice to have.

Reply:
We want to make fleet-server unhealthy if it can't connect to remote ES, but we don't want it to go offline. Perhaps the client should be created conditionally, or catch the error if it can't be created.

Reply:
Usually if you just do a GET to the ES hostname it should give a response with the version and some other things. I think this works for Serverless too although that may have changed since I last tried it.

Reply:
that's good for testing availability, though doesn't verify if the provided service_token is valid to create api keys for agents

Reply:
I added logic locally to ping every remote es output host every 5s in self monitor, but I'm not sure if it's the best thing to do, fleet-server would go to degraded even though the inaccessible remote ES is not used by any agents.
Shouldn't we only ping those hosts that are being used? E.g. we could ping on agent checkin.